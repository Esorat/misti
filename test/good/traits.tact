trait T {
  abstract get fun getter(): Int;
  abstract const something: Int;
}

contract TestContract with T {
  override get fun getter(): Int { return 0 }
  override const something: Int = 42;
}

trait V {
  a: Int;
  get fun getter(): Int {
      return self.a + 1; // a is used
  }
}

contract TestContract with V {
    a: Int = 42;
    fun test() {}
}

trait TParent {
  a: Int;
  fun getA(): Int {
      return self.a; // a is used
  }
}

trait TChild with TParent {
  a: Int;
}

contract TestContract with TChild {
    a: Int = 19;
    fun test(): Int {
        return self.getA();
    }
}