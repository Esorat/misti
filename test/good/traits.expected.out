Misti execution failed:
Compilation error: ../../../../var/folders/vp/kzqvq7715ln9r_vglmcbnw980000gn/T/misti/temp-NtWVen/test/good/traits.tact:18:1: Type "TestContract" already exists
Line 18, col 1:
  17 | 
> 18 | contract TestContract with V {
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  19 |     a: Int = 42;
